boot/freeldr/freeldr/arch/i386/entry.S

```diff
@@ -47,6 +48,11 @@ _RealEntryPoint:
     mov gs, ax
     mov ss, ax
 
+    /* Keep only the low 16 bits of stack */
+    and esp, HEX(FFFF)
```

Comment:
What would happen if you were to keep esp as it originally was?

Reply:
when you switch from real mode to protected/long mode you can't assume that the high 16 bits of `esp/rsp` is clean.

Reply:
> What would happen if you were to keep esp as it originally was?

triple fault.